### PR TITLE
[#2127] Improvement(core): improve TreeLock for ease of unlock and fix unexpected err when lock failed

### PR DESCRIPTION
**Pull request by YxAc**
### What changes were proposed in this pull request?
- improve TreeLock for ease of unlock by keeping lock type
- fix unexpected err when lock failed and unlock all nodes that have been locked

### Why are the changes needed?
Fix: #2127 

### Does this PR introduce _any_ user-facing change?
N/A

### How was this patch tested?
UT

### CODE REVIEW

**Comment by yuqi1129** [comment]
@YxAc 
Can you add some tests about the scenario when the method `lock` fails (throws exceptions) and everything is still okay, that is:  we can unlock the nodes that have been locked. You can mock the `lock` method and do some corresponding tests for it. There are some similar tests in the `TestLockManager`, you can refer to them.

**Comment by YxAc** [comment]
> @YxAc Can you add some tests about the scenario when the method `lock` fails (throws exceptions) and everything is still okay, that is: we can unlock the nodes that have been locked. You can mock the `lock` method and do some corresponding tests for it. There are some similar tests in the `TestLockManager`, you can refer to them.

done

**Review by yuqi1129 — DISMISSED** [review]
It's okay with me, @jerryshao . Could you spare some time to take a look?

File: core/src/main/java/com/datastrato/gravitino/lock/TreeLock.java [hunk 1]
```diff
@@ -84,8 +87,21 @@ public void lock(LockType lockType) {
     for (int i = 0; i < length; i++) {
       TreeLockNode treeLockNode = lockNodes.get(i);
       LockType type = i == length - 1 ? lockType : LockType.READ;
-      treeLockNode.lock(type);
-      heldLocks.push(treeLockNode);
+      try {
+        treeLockNode.lock(type);
+        heldLocks.push(Pair.of(treeLockNode, type));
+        LOG.trace("Locked node: {}, lock type: {}", treeLockNode, type);
```
Comment (jerryshao) [inline core/src/main/java/com/datastrato/gravitino/lock/TreeLock.java:93]: Shall we also add this to like `if (LOG.isTraceEnabled())`?

Comment (yuqi1129) [reply core/src/main/java/com/datastrato/gravitino/lock/TreeLock.java:93]: The log message's logic "if(Log.isTrace())") will be helpful if it has complex compute/IO logic or string concat, or if it doesn't have much affection.

Comment (YxAc) [reply core/src/main/java/com/datastrato/gravitino/lock/TreeLock.java:93]: Yes, I also consider that this log does not have much overhead, such as string concat, so didn't add judgment statement
I can add this to keep consistent.

**Review by jerryshao — APPROVED** [review]
LGTM, Thanks @YxAc for your contribution.

**Pull request merged**